src/test/java/org/elasticsearch/test/ElasticsearchTestCase.java

```diff
@@ -213,6 +214,20 @@ public void ensureAllArraysReleased() throws Exception {
         MockBigArrays.ensureAllArraysAreReleased();
     }
 
+    @After
+    public void ensureAllSearchContextsReleased() throws Exception {
+        awaitBusy(new Predicate<Object>() {
```

Comment:
can we use `assertBusy` here? so we have the exact count at the time of failure?